src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,103 @@
+});
+
+const addBook = createAsyncThunk('AddBook', async (book) => {
+  const response = await axios.post(API_URL, book);
+  return response.data === 'Created' ? book : null;
+});
+
+const removeBook = createAsyncThunk('RemoveBook', async (ITEM_ID) => {
+  const response = await axios.delete(`${API_URL}/${ITEM_ID}`);
+  return response.data === 'The book is removed!' ? ITEM_ID : null;
+});
+
+// INITIAL STATE
+const initialState = {
+  books: [],
+  error: '',
+  status: 'idle',
+};
+
+// SLICE
+export const booksSlice = createSlice({
   name: 'books',
-  initialState: initialBooksState,
-  reducers: {
-    addBook: (state, action) => {
-      state.push(action.payload);
-    },
-    removeBook: (state, action) => (
-      state.filter((book) => book.item_id !== action.payload)
-    ),
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchBooks.pending, (state) => {
+        state.status = 'loading';
+      })
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        state.status = 'fetched';
+        if (action.payload !== '') {
+          const books = [];
+          const keys = Object.keys(action.payload);
+          keys.forEach((Id) => {
+            books.push({ item_id: Id, ...action.payload[Id][0] });
+          });
+          state.books = books;
+          if (state.books.length === 0) {
+            state.error = 'No books found';
+          }
+        } else {
+          state.error = 'No books found';
+        }
+      })
+      .addCase(fetchBooks.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      });
+
+    builder
+      .addCase(addBook.pending, (state) => {
+        state.status = 'pending';
+      })
+      .addCase(addBook.fulfilled, (state, action) => {
+        if (action.payload !== null) {
+          state.status = 'succeeded';
+          state.error = '';
+          state.books.push(action.payload);
+        } else {
+          state.status = 'failed';
+          state.error = 'Unable to add book';
+        }
+      })
+      .addCase(addBook.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      });
+
+    builder
+      .addCase(removeBook.pending, (state) => {
+        state.status = 'loading';
+      })
+      .addCase(removeBook.fulfilled, (state, action) => {
+        if (action.payload !== null) {
+          state.status = 'succeeded';
+          state.error = '';
+          state.books = state.books.filter((bookId) => bookId.item_id !== action.payload);
+          if (state.books.length === 0) state.error = 'No book was found!';
+        } else {
+          state.status = 'failed';
+          state.error = 'Unable to remove book';
+        }
+      })
+      .addCase(removeBook.rejected, (state, action) => {
+        state.status = 'failed';
+        state.error = action.error.message;
+      });
   },
 });
 
```

Comment:
![image](https://github.com/HossainAraf/Book-store/assets/93039359/a5a83b8a-5966-48ff-8cb3-aa38311e92c4)

**[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be nice if you fix these issues we are facing on the console screen because these can break your app entirely :fearful: Apologies on behalf of the previous reviewer that he didn't mention these things but it would be nice if you take this matter into your consideration :blush: